packages/suite/src/hooks/wallet/useSendFormCompose.ts

```diff
@@ -159,13 +190,41 @@ export const useSendFormCompose = ({
             const { setMaxOutputId } = values;
             // set calculated and formatted "max" value to `Amount` input
             if (typeof setMaxOutputId === 'number' && composed.max) {
-                setAmount(setMaxOutputId, composed.max);
+                const networkDecimals = NETWORKS.find(network => network.symbol === account.symbol)
+                    ?.decimals as number;
+
+                setAmount(
+                    setMaxOutputId,
+                    areSatsUsed ? amountToSatoshi(composed.max, networkDecimals) : composed.max,
+                );
                 setDraftSaveRequest(true);
             }
         },
-        [composeField, getValues, setAmount, errors, setError, clearErrors, setValue],
+        [
+            composeField,
+            getValues,
+            setAmount,
+            errors,
+            setError,
+            clearErrors,
+            setValue,
+            areSatsUsed,
+            account,
+        ],
     );
 
+    useDidUpdate(() => {
```

Comment:
possible race condition
im not sure that it will not break something. take a note that useSendForm > useFormDraft is called after mount.
loading form from draft is an async process (mount > TrezorConnect.compose > processing data > filling the form)